File: tests/leap_compat/swift/TypedStreamConsumer.swift
import LeapSDK

func requireNonthrowing<S: AsyncSequence>(_ sequence: S) where S.Failure == Never {}

func typedStream(_ conversation: any Conversation) -> SkieSwiftFlow<any MessageResponse> {
  let responses: SkieSwiftFlow<any MessageResponse> = conversation.generateResponse(
    userTextMessage: "hello", generationOptions: nil)
  requireNonthrowing(responses)
  return responses
}

File: tests/leap_compat/kotlin/TypedStreamConsumer.kt
package consumer

import ai.liquid.leap.Conversation
import ai.liquid.leap.message.MessageResponse
import kotlinx.coroutines.flow.Flow

fun typedStream(conversation: Conversation): Flow<MessageResponse> =
    conversation.generateResponse(userTextMessage = "hello", generationOptions = null)

fun isTerminal(response: MessageResponse): Boolean =
    when (response) {
        is MessageResponse.Complete, is MessageResponse.Error -> true

        is MessageResponse.Chunk, is MessageResponse.ReasoningChunk,
        is MessageResponse.AudioSample, is MessageResponse.FunctionCalls,
        -> false
    }
